src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -307,7 +307,7 @@ def convert_sf_to_sp_type(
     if column_type_name == "REAL":
         return DoubleType()
     if (column_type_name == "FIXED" or column_type_name == "NUMBER") and scale == 0:
-        return LongType()
+        return LongType(precision=precision)
```

Comment:
if we make the argument private, it will be:
```suggestion
        return LongType(_precision=precision)
```